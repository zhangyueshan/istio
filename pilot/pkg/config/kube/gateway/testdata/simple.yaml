apiVersion: networking.x-k8s.io/v1alpha1
kind: GatewayClass
metadata:
  name: istio
spec:
  controller: istio.io/gateway-controller
---
apiVersion: networking.x-k8s.io/v1alpha1
kind: Gateway
metadata:
  name: gateway
  namespace: istio-system
spec:
  class: istio
  listeners:
  - hostname:
      match: Domain
      name: domain.example
    port: 80
    protocol: HTTP
    routes:
      routeNamespaces: {}
      routeSelector:
        matchLabels:
          selected: "yes"
      group: ""
      resource: httproutes
  - hostname:
      match: Any
    port: 34000
    protocol: TCP
    routes:
      routeNamespaces: {}
      routeSelector:
        matchLabels:
          selected: "yes"
      resource: tcproutes
---
apiVersion: networking.x-k8s.io/v1alpha1
kind: HTTPRoute
metadata:
  name: http
  namespace: default
  labels:
    selected: "yes"
spec:
  hosts:
  - hostnames: ["first.domain.example", "another.domain.example"]
    rules:
    - match:
        pathMatchType: Prefix
        path: /get
        headers:
          my-header: some-value
      filter:
        headers:
          add:
            my-added-header: added-value
          remove: [my-removed-header]
      action:
        forwardTo:
        - targetRef:
            name: httpbin
  - hostnames: ["second.domain.example"]
    rules:
    - match:
        pathMatchType: Prefix
        path: /second
      action:
        forwardTo:
        - targetRef:
            name: httpbin-second
---
apiVersion: networking.x-k8s.io/v1alpha1
kind: HTTPRoute
metadata:
  name: http-not-selected
  namespace: default
  labels:
    selected: "nope"
spec:
  hosts:
  - hostnames: ["should.not.select"]
    rules:
    - match:
        pathMatchType: Prefix
        path: /get
      action:
        forwardTo:
        - targetRef:
            name: httpbin
---
apiVersion: networking.x-k8s.io/v1alpha1
kind: TCPRoute
metadata:
  name: tcp
  namespace: default
  labels:
    selected: "yes"
spec:
  rules:
  - action:
      forwardTo:
        targetRef:
          name: httpbin
