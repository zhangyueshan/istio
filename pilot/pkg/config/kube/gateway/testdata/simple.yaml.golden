apiVersion: networking.istio.io/v1alpha3
kind: Gateway
metadata:
  creationTimestamp: null
  name: gateway-istio-autogenerated-k8s-gateway
  namespace: istio-system
spec:
  selector:
    istio: ingressgateway
  servers:
  - hosts:
    - '*.domain.example'
    port:
      name: http-80-gateway-gateway-istio-system
      number: 80
      protocol: HTTP
  - hosts:
    - '*'
    port:
      name: tcp-34000-gateway-gateway-istio-system
      number: 34000
      protocol: TCP
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  creationTimestamp: null
  name: tcp-tcp-istio-autogenerated-k8s-gateway
  namespace: default
spec:
  gateways:
  - istio-system/gateway-istio-autogenerated-k8s-gateway
  hosts:
  - '*'
  tcp:
  - route:
    - destination:
        host: httpbin.default.svc.cluster.local
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  creationTimestamp: null
  name: http-0-istio-autogenerated-k8s-gateway
  namespace: default
spec:
  gateways:
  - istio-system/gateway-istio-autogenerated-k8s-gateway
  hosts:
  - first.domain.example
  - another.domain.example
  http:
  - headers:
      request:
        add:
          my-added-header: added-value
        remove:
        - my-removed-header
    match:
    - headers:
        my-header:
          exact: some-value
      uri:
        prefix: /get
    route:
    - destination:
        host: httpbin.default.svc.cluster.local
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  creationTimestamp: null
  name: http-1-istio-autogenerated-k8s-gateway
  namespace: default
spec:
  gateways:
  - istio-system/gateway-istio-autogenerated-k8s-gateway
  hosts:
  - second.domain.example
  http:
  - match:
    - uri:
        prefix: /second
    route:
    - destination:
        host: httpbin-second.default.svc.cluster.local
---
